### PR TITLE
[autoscaler] Add an initial_workers option

### PR DESCRIPTION
**Pull request by mattearllongshot**
## What do these changes do?

    This option goes along with `min_workers`, and `max_workers`.  When the
    cluster is first brought up (or when it is refreshed with a subsequent
    `ray up`) this number of nodes will be started.
    
    It's a workaround for issues of scaling (see related issues) where it
    can take a long time (or forever in the case where the head node has
    `--num-cpus 0`) to scale up a cluster in response to increasing demand.

## Related issue number

Workaround for https://github.com/ray-project/ray/issues/3339 and https://github.com/ray-project/ray/issues/2106

### CODE REVIEW

**Comment by ericl** [comment]
The implementation looks. Could we add a unit test?

File: python/ray/autoscaler/autoscaler.py [hunk 1]
```diff
@@ -467,10 +473,15 @@ def reload_config(self, errors_fatal=False):
                 logger.exception("StandardAutoscaler: Error parsing config.")
 
     def target_num_workers(self):
-        target_frac = self.config["target_utilization_fraction"]
-        cur_used = self.load_metrics.approx_workers_used()
-        ideal_num_nodes = int(np.ceil(cur_used / float(target_frac)))
-        ideal_num_workers = ideal_num_nodes - 1  # subtract 1 for head node
+        initial_workers = self.config.get("initial_workers", 0)
```
[the remark this answers is not in the document]
Comment (ls-daniel) [reply python/ray/autoscaler/autoscaler.py]: Should we change the schema to reflect this?

At the moment, various things are marked as OPTIONAL e.g. min_workers, target_utilization_fraction, and so on; but they're hardly optional if the autoscaler will explode without them being specified.

**Comment by ls-daniel** [comment]
I've added initial_workers to default-full.yaml here and written a unit test.
I can't commit to this pull request so I'll hopefully be able to work that out with @mattearllongshot over the next few days.

**Comment by mattearllongshot** [comment]
Is everyone happy to merge this now?

**Review by ericl — APPROVED** [review]

File: test/autoscaler_test.py [hunk 2]
```diff
@@ -314,6 +315,27 @@ def testDynamicScaling(self):
...
+        autoscaler = StandardAutoscaler(
+            config_path,
+            LoadMetrics(),
+            max_launch_batch=5,
+            max_concurrent_launches=5,
+            max_failures=0,
+            update_interval_s=0)
+        self.waitForNodes(0)
+        autoscaler.update()
+        self.waitForNodes(5) # expected due to batch sizes and concurrency
```
Comment (richardliaw) [inline test/autoscaler_test.py]:
```suggestion
        self.waitForNodes(5)  # expected due to batch sizes and concurrency
```

Comment (richardliaw) [inline test/autoscaler_test.py]:
```suggestion
        self.waitForNodes(5)  # expected due to batch sizes and concurrency
```

Comment (richardliaw) [reply test/autoscaler_test.py]: (lint)

**Comment by richardliaw** [comment]
jenkins retest this please

**Review by richardliaw — APPROVED** [review]

**Comment by richardliaw** [comment]
@mattearllongshot, @ls-daniel, thanks for contributing this; sorry for the delay in merging!

**Pull request merged**